controller_manager/include/controller_manager/controller_manager.hpp

```diff
@@ -390,6 +390,16 @@ class ControllerManager : public rclcpp::Node
     const std::vector<ControllerSpec> & controllers, int strictness,
     const ControllersListIterator controller_it);
 
+  /// Sets if the parsed vector of controller's reference interfacs are available to use or not
+  /**
+   * Sets the availability status of the reference interfaces of the given list of controllers
+   * @param[in] controllers list
+   * @param[in] available - To specify they are needed to be available or unavailable for other
+   * controllers to use
+   */
+  void set_controllers_reference_interfaces_availability(
```

Comment:
Yes ofcourse. The idea is the same to have it available always, but the problem here is when the state interfaces are being used the controller should report it as it being in the chained mode or not. I thought it would make sense for it to be reported chained mode but then we have a separate method to manage the switching of references between subscribers or interfaces